src-ui/src/app/components/document-list/document-list.component.ts

```diff
@@ -101,11 +110,62 @@ export class DocumentListComponent implements OnInit, OnDestroy {
           this.list.reload()
           this.unmodifiedFilterRules = view.filter_rules
         })
-      } else {
+      })
+
+    const filterQueryVars: string[] = FILTER_RULE_TYPES.map(
+      (rt) => rt.filtervar
+    )
+
+    this.route.queryParamMap
+      .pipe(filter((qp) => !this.route.snapshot.paramMap.has('id'))) // only when not on saved view
+      .subscribe((queryParams) => {
+        const queryVarsFilterRules = []
+
+        filterQueryVars.forEach((filterQueryVar) => {
```

Comment:
maybe you can extract this in a method with a clear name. It is hard to understand, what is happening here.

queryVarsFilterRules.push(`whateverIsHappeningHere(queryParams)`)

Reply:
Cheers, always appreciate your thoughts. I shrunk this logic down and hopefully made it much more readable. Its transforming query vars into "filter rules"